src/OrleansRuntime/Core/InsideRuntimeClient.cs

```diff
@@ -336,6 +341,15 @@ public void SniffIncomingMessage(Message message)
                 try
                 {
                     var request = (InvokeMethodRequest) message.BodyObject;
+                    if (request.Arguments != null)
+                    {
+                        for (var i = 0; i < request.Arguments.Length; i++)
+                        {
+                            var arg = request.Arguments[i];
+                            if (!(arg is GrainCancellationToken)) continue;
```

Comment:
There should be a comment here explaining what's going on. We are using a surrogate type for serialization. Is this something we should add proper support for? You can imagine we might want a similar surrogate type for IObservable<T>, for example.